dbms/src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -3807,4 +3830,88 @@ bool MergeTreeData::moveParts(CurrentlyMovingPartsTagger && moving_tagger)
     return true;
 }
 
+void MergeTreeData::loadPartsInsertionHistory()
```

Comment:
BTW, the history is not partitioned. It means that (in contrast to ReplicatedMergeTree), we cannot insert the same data after DROP partition.

DROP and reinsert is popular use case.

Reply:
done